examples/drupal_loadtest.rs

```diff
@@ -101,23 +101,23 @@ fn drupal_loadtest_front_page(client: &mut GooseClient) {
 }
 
 /// View a node from 1 to 10,000, created by preptest.sh.
-fn drupal_loadtest_node_page(client: &mut GooseClient) {
+async fn drupal_loadtest_node_page<'r>(client: &'r mut GooseClient) {
```

Comment:
Maybe we want to add a macro for that as well - it’s always the same:

- `goose_fn! drupal_loadtest_node_page() {
}`

is what I would suggest.

Reply:
I made a sketch of an attribute style macro in a PR against this branch: https://github.com/jeremyandrews/goose/pull/1
This would make it look like:
```
#[goose_client_callback]
async fn website_login<'r>(client: &'r mut GooseClient) {
    ...
}
```

It may be possible to cut out the arguments and lifetime annotations, but it seems too magical so I'd rather not. Most people will be happy to copy-paste from the examples, so as long as the error messages are easy to understand, I think the verbosity is probably fine.

Honestly, I think of both approaches as temporary workarounds until we can fix the language to be more expressive around fn lifetimes (at which point we can make ::new() accept a bare async fn directly).

Reply:
Given the lifetime is not needed, this might still be okay, but not as necessary now.